Sources/SwiftSyntaxBuilderGeneration/Templates/BuildableCollectionNodesFile.swift

```diff
@@ -0,0 +1,257 @@
+        }
+      }
+
+      FunctionDecl(
+        modifiers: [TokenSyntax.public],
+        identifier: .identifier("build\(type.baseName)"),
+        signature: FunctionSignature(
+          input: createFormatLeadingTriviaParameters(withDefaultTrivia: true),
+          output: type.syntax
+        )
+      ) {
+        VariableDecl(letOrVarKeyword: .let) {
+          PatternBinding(
+            pattern: "result",
+            initializer: InitializerClause(
+              value: FunctionCallExpr(MemberAccessExpr(base: "SyntaxFactory", name: "make\(type.baseName)")) {
+                if elementType.isToken {
+                  TupleExprElement(expression: "elements")
+                } else {
+                  TupleExprElement(
+                    expression: FunctionCallExpr(MemberAccessExpr(base: "elements", name: "map"),
+                    trailingClosure: ClosureExpr {
+                      FunctionCallExpr(MemberAccessExpr(base: "$0", name: "build\(elementType.baseName)")) {
+                        TupleExprElement(label: "format", expression: "format")
+                        TupleExprElement(
+                          label: "leadingTrivia",
+                          expression: node.elementsSeparatedByNewline
+                            ? SequenceExpr(elements: ExprList([
+                                MemberAccessExpr(base: "Trivia", name: "newline"),
+                                BinaryOperatorExpr("+"),
+                                FunctionCallExpr(MemberAccessExpr(base: "format", name: "_makeIndent")),
+                              ]))
+                            : "nil"
+                        )
+                      }
+                    })
+                  )
+                }
+              }
+            )
+          )
+        }
+        IfStmt(
+          conditions: ConditionElementList([
+            ConditionElement(condition: OptionalBindingCondition(
+              letOrVarKeyword: .let,
+              pattern: "leadingTrivia",
+              initializer: InitializerClause(value: "leadingTrivia")
+            ))
+          ]),
+          body: CodeBlock {
+            ReturnStmt(expression: FunctionCallExpr(MemberAccessExpr(base: "result", name: "withLeadingTrivia")) {
+              SequenceExpr(elements: ExprList([
+                "leadingTrivia",
+                BinaryOperatorExpr("+"),
+                TupleExpr {
+                  SequenceExpr(elements: ExprList([
+                    MemberAccessExpr(base: "result", name: "leadingTrivia"),
+                    BinaryOperatorExpr("??"),
+                    ArrayExpr(elements: [])
+                  ]))
+                }
+              ]))
+            })
+          },
+          elseKeyword: .else,
+          elseBody: CodeBlock {
+            ReturnStmt(expression: "result")
+          }
+        )
+      }
+
+      FunctionDecl(
+        modifiers: [TokenSyntax.public],
+        identifier: .identifier("buildSyntax"),
```

Comment:
We are using `TokenSyntax.identifier(...)` in a lot of places. This got me thinking whether we might be able to conform `TokenSyntax` to `ExpressibleByStringLiteral`?

Reply:
-> Moving the discussion to #507.